### PR TITLE
Update controller_revision class constructor, documentation and add 'to_dict' parameters validation

### PR DESCRIPTION
**Pull request by oharan2**
Main changes:

1. Added support for passing `revision` arg for the ControllerRevision instructor class
- For being a mandatory arg, we validate revision being passed and if not, raise
2. Use super class constructor with kwargs to pass Resource instance args

### CODE REVIEW

**Comment by oharan2** [comment]
@myakove Can you please indicate if there's something I should fix regarding the failing checks? everything is working good locally.

File: pyproject.toml [hunk 1]
```diff
@@ -47,11 +47,12 @@ requires = ["poetry-core>=1.0.0", "poetry-dynamic-versioning"]
 build-backend = "poetry_dynamic_versioning.backend"
 
 [tool.poetry.dependencies]
-python = "^3.8"
+python = ">=3.9,<3.11"
```
Comment (omrirh) [inline pyproject.toml:50]: maybe this fails some of the PR checks?
see [failing logs](https://qaz.myakove.com/webhook_server/python-module-install/openshift-python-wrapper-PR-1528-cf6f7379a3e602e66da204f8263568f10c966c5f) saying `ERROR: Package 'openshift-python-wrapper' requires a different Python: 3.11.6 not in '<3.11,>=3.9'`

Comment (oharan2) [reply pyproject.toml:50]: So is tox failing simply because `tox.ini` file doesn't specify a valid ('<3.11,>=3.9') Pyhton version?

Comment (rnetser) [reply pyproject.toml:50]: you cannot remove pythonh3.8 support from the project; find numpy release which does work with 3.8

**Review by rnetser — CHANGES_REQUESTED** [review]
Please update commit message (pr description)

**Comment by oharan2** [comment]
Hi @myakove the tox check is currently failing with `detached HEAD` mode,
requires to set `advice.detachedHead` to false in the project git config, or is it something in my pr?
tox output- https://qaz.myakove.com/webhook_server/tox/openshift-python-wrapper-PR-1528-011487e3ac42f2f469ff345cfd95f7fbaa067aec
Thanks

File: ocp_resources/controller_revision.py [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
         self.owner_references = owner_references
         self.revision_object = revision_object
+        self.revision = revision
 
     def to_dict(self):
         super().to_dict()
         if not self.yaml_file:
+            if not self.revision:
+                raise ValueError("Passing yaml_file or parameter 'revision' is required.")
+            if not isinstance(self.revision, numpy.int64):
```
Comment (myakove) [inline ocp_resources/controller_revision.py:36]: please remove this check, we do not check the user type
We let the openshift API do it and fail if needed

**Review by myakove — CHANGES_REQUESTED** [review]

**Review by rnetser — APPROVED** [review]

**Review by rnetser — APPROVED** [review]

**Review by myakove — APPROVED** [review]

**Pull request merged**